### PR TITLE
feat: Make MFA instructions clear for SMS and Email

### PR DESCRIPTION
**Pull request by maknahar**
For Email and SMS based MFA:

Before:

Screenshot 2024-10-10 at 3 30 52 PM

After:

Screenshot 2024-10-10 at 4 49 00 PM

No change for TOTP based MFA.

### CODE REVIEW

**Comment by hsluoyz** [comment]
@dacongda plz review

**Review by dacongda — APPROVED** [review]

**Comment by maknahar** [comment]
@hsluoyz Looks like you closed this PR after asking for approval. I am a little confused. Any reason for not merging the PR?

File: web/src/auth/mfa/MfaAuthVerifyForm.js [hunk 1]
```diff
@@ -71,20 +71,29 @@ export function MfaAuthVerifyForm({formValues, authParams, mfaProps, application
...
-            application={application}
-          />) : (
-          <MfaVerifyTotpForm
-            mfaProps={mfaProps}
-            onFinish={verify}
-          />
+          <div>
+            <div style={{marginBottom: 24}}>
+              {i18next.t(
+                "mfa:You have enabled Multi-factor Authentication. Please click Send Code and follow the prompts to finish your login process"
```
Comment (hsluoyz) [inline web/src/auth/mfa/MfaAuthVerifyForm.js:78]: Use: `"mfa:You have enabled Multi-Factor Authentication. Please click "Send Code" to continue"`

File: web/src/auth/mfa/MfaAuthVerifyForm.js [hunk 2]
```diff
@@ -71,20 +71,29 @@ export function MfaAuthVerifyForm({formValues, authParams, mfaProps, application
...
+              method={mfaAuth}
+              onFinish={verify}
+              application={application}
+            />
+          </div>
+        ) : (
+          <div>
+            <div style={{marginBottom: 24}}>
+              {i18next.t(
+                "mfa:You have enabled multi-factor authentication, please enter the authentication code"
```
Comment (hsluoyz) [inline web/src/auth/mfa/MfaAuthVerifyForm.js:92]: Use: `"mfa:You have enabled Multi-Factor Authentication, please enter the TOTP code"`

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: web/src/auth/mfa/MfaAuthVerifyForm.js (hunk 1 above)
Comment (maknahar) [reply web/src/auth/mfa/MfaAuthVerifyForm.js:78]: Make sense. I will make these changes and re-request the review.

File: web/src/auth/mfa/MfaAuthVerifyForm.js (hunk 2 above)
Comment (maknahar) [reply web/src/auth/mfa/MfaAuthVerifyForm.js:92]: This sounds better to me as well. However, this is an existing message so I hope it is okay if I update it.

Comment (maknahar) [reply web/src/auth/mfa/MfaAuthVerifyForm.js:92]: As I am updating the existing message, there are a couple of more minor enhancements in MFA dialog that I see and I will make those changes as well.

1. Changing "Multi-falcon authentication" to "Multi-factor Authentication" in the heading. 
2. Adding a space between `Having a problem` and `Use recovery code`

Please let me know if these two sounds like a good improvement to you @hsluoyz @dacongda

Comment (hsluoyz) [reply web/src/auth/mfa/MfaAuthVerifyForm.js:92]: Both OK

File: web/src/auth/mfa/MfaAuthVerifyForm.js [hunk 3]
```diff
@@ -24,7 +24,14 @@ import MfaVerifyTotpForm from "./MfaVerifyTotpForm";
 export const NextMfa = "NextMfa";
 export const RequiredMfa = "RequiredMfa";
 
-export function MfaAuthVerifyForm({formValues, authParams, mfaProps, application, onSuccess, onFail}) {
+export function MfaAuthVerifyForm({
```
Comment (dacongda) [inline web/src/auth/mfa/MfaAuthVerifyForm.js:27]: plz do not change code format

Comment (maknahar) [reply web/src/auth/mfa/MfaAuthVerifyForm.js:27]: @dacongda These seems to be a commit hook configured in repo which runs `yarn run fix` causing these changes. I will reset these changes in the last commit though.

**Comment by dacongda** [comment]
we will try to pull a new pr later

**Comment by hsluoyz** [comment]
This PR is becoming too tedious compared to its contribution size (aka too difficult, but contribution is small). Closed for now.

**Comment by maknahar** [comment]
@hsluoyz @dacongda I have spent too much time on this PR for it to be closed now. Pr has become big because you asked for the change in the existing messages and I explicitly told that is the case. 😞

**Comment by maknahar** [comment]
Also, contribution might be "Small" but it is very essential for any production system to avoid creating confusion among the users using MFA.

**Comment by hsluoyz** [comment]
@maknahar I totally understand what you said and your desire to have your hand-made patch merged but it seems that it will take us less time to fix it by ourselves than reviewing your PR.

**Comment by dacongda** [comment]
> Also, contribution might be "Small" but it is very essential for any production system to avoid creating confusion among the users using MFA.

I know it , and I will fix this in few minute, sorry for spending you too much time on this.😂

**Comment by maknahar** [comment]
Okay. Thanks for the quick PR for the fix.

**Pull request closed without merging**